### PR TITLE
refactor ROCR documentation

### PR DESCRIPTION
**Pull request by SwRaw**
Refactoring all libraries to adhere to the documentation standards that includes following home page format, index style, language improvements, link fixes, navigation, and overall content readability improvements.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+- A 64-bit version of AMD’s runtime image library
+
+The contents of these packages are installed in ``/opt/rocm/hsa`` and ``/opt/rocm`` by default. The core runtime package depends on the ``hsakmt-roct-dev`` package.
+
+Build and install from source
+--------------------------------
+
+Here is how you can build ROCR from source.
+
+Prerequisites
```
Comment (SwRaw) [inline docs/installation.rst:49]: @dayatsin-amd Please add commands where applicable under Prerequisites.

File: docs/installation.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+====================
+
+This document provides information required to build and install ROCR using prebuilt binaries or from source.
+
+Build and install using prebuilt binaries
+-------------------------------------------
+
+Here is how you can install ROCR using prebuilt binaries.
+
+Prerequisites
```
Comment (SwRaw) [inline docs/installation.rst:18]: @dayatsin-amd Add commands wherever applicable under Prerequisites.

**Review by lpaoletti — COMMENTED** [review]
This item requires a redirect:
image

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    * :ref:`installation`
+    
+  .. grid-item-card:: API reference
+
+    * :ref:`c-interface-adaptors`
+    * :ref:`environment-variables`
+    * :ref:`api`
+
+  .. grid-item-card:: Contribution
```
Comment (lpaoletti) [inline docs/index.rst:30]: I don't believe this is a standard box, and it should be removed. Here and from the TOC

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    * :ref:`contributing-to-rocr`
+
+To contribute to ROCR, refer to :ref:`contributing-to-rocr`.
```
Comment (lpaoletti) [inline docs/index.rst:34]: Isn't this a duplicate of what is on top of the boxes?

File: docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    :description: HSA runtime implementation
+    :keywords: ROCR, ROCm, library, tool, runtime
+
+.. _index:
+
+=====================
+ROCR documentation
+=====================
+
+The ROCm runtime (ROCR) is AMD's implementation of HSA runtime, which is a thin, user-mode API that exposes the necessary interfaces to access and interact with graphics hardware driven by the AMDGPU driver set and the ROCK kernel driver. To learn more, see :ref:`what-is-rocr-runtime`
```
Comment (lpaoletti) [inline docs/index.rst:11]: this link is broken:
image

File: docs/environment_variables.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
+.. meta::
+   :description: HSA runtime implementation
+   :keywords: ROCR, ROCm, library, tool, runtime
+
+.. _environment-variables:
+
+Environment variables
+========================
+
+The following table lists the useful environment variables.
```
Comment (lpaoletti) [inline docs/environment_variables.rst:10]: The term "useful" should be removed. 
it means that other variables are useless. 

Are these the only variables? if not then say something like, 
The following table lists the most often used variables.

File: docs/environment_variables.rst [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    
+    * - Environment Variable
+      - Possible values
+      - Description
+
+    * - HSA_ENABLE_SDMA
+      - 
+        * 0: Disabled
+        * 1: Enabled (default)
+      - This enables or disables the use of DMA engines in all copy directions (Host-to-Device, Device-to-Host, Device-to-Device) when using the
```
Comment (lpaoletti) [inline docs/environment_variables.rst:23]:
```suggestion
      - Specifies the use of DMA engines in all copy directions (Host-to-Device, Device-to-Host, Device-to-Device) when using the
```

File: docs/environment_variables.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        ``hsa_memory_copy``, ``hsa_amd_memory_fill``, ``hsa_amd_memory_async_copy``, ``hsa_amd_memory_async_copy_on_engine`` APIs
+
+    * - HSA_ENABLE_PEER_SDMA
+      -
+        * 0: Disabled
+        * 1: Enabled (default)
+      - This enables or disables the use of DMA engines for Device-to-Device copies when using the ``hsa_memory_copy``, ``hsa_amd_memory_async_copy``, ``hsa_amd_memory_async_copy_on_engine`` APIs
```
Comment (lpaoletti) [inline docs/environment_variables.rst:30]:
```suggestion
      - Specifies the use of DMA engines for Device-to-Device copies when using the ``hsa_memory_copy``, ``hsa_amd_memory_async_copy``, ``hsa_amd_memory_async_copy_on_engine`` APIs
```

File: docs/c-interface-adaptors.rst [hunk 9]
```diff
@@ -0,0 +1,16 @@
+.. meta::
+   :description: HSA runtime implementation
+   :keywords: ROCR, ROCm, library, tool, runtime
+
+.. _c-interface-adaptors:
+
+C Interface Adaptors
+=====================
+
+The C interface layer is the :ref:`top layer in ROCR <runtime-design>` that provides C++ APIs as defined in the `HSA Runtime Specification 1.2 <https://hsafoundation.com/wp-content/uploads/2021/02/HSA-Runtime-1.2.pdf>`_. The C interface layer also consists of the interfaces and default definitions for the standard extensions. The interface functions simply forward to a function pointer table defined here. The table is initialized to point to default definitions, which simply returns an appropriate error code. If available, the extension library is loaded as part of runtime initialization and the table is updated to point to the extension library.
```
Comment (lpaoletti) [inline docs/c-interface-adaptors.rst:10]: Don't understand where the top layer link is supposed to take you

File: docs/contributing-to-rocr.rst [hunk 10]
```diff
@@ -0,0 +1,93 @@
+.. meta::
```
Comment (lpaoletti) [inline docs/contributing-to-rocr.rst:1]: I don't recall seeing a contributing topic in the other refactored projects - why do we need this one?

File: docs/index.rst (hunk 3 above)
Comment (SwRaw) [reply docs/index.rst:30]: @lpaoletti We have contribution in many other components as well. Refer MIGraphX, hipTensor, hipBLASLt to name a few. It is also required as a lot of information applies to contributors and not to the users. It is better to segregate information according to the type of audience.

File: docs/index.rst (hunk 4 above)
Comment (SwRaw) [reply docs/index.rst:34]: Yes, last time I, you and Randy had a discussion about confusion regarding the contributing to docs or the component and you had suggested adding this line.

File: docs/environment_variables.rst (hunk 7 above)
Comment (SwRaw) [reply docs/environment_variables.rst:23]: @lpaoletti I guess "control" would be a better word. Updated it.

File: docs/c-interface-adaptors.rst (hunk 9 above)
Comment (SwRaw) [reply docs/c-interface-adaptors.rst:10]: @lpaoletti It takes you to the Runtime Design section which talks about the three layers. I had to separate out the top layer as it concerns the API users while the bottom two layers concern with the contributors.

File: docs/contributing-to-rocr.rst (hunk 10 above)
Comment (SwRaw) [reply docs/contributing-to-rocr.rst:1]: @lpaoletti It is there Leo. Refer to hipBLASlt, MIgraphX, hipTensor etc.

**Comment by SwRaw** [comment]
> This item requires a redirect: image

I will handle the redirects once the PR is merged so that I can provide the exact URLs.

File: docs/environment_variables.rst (hunk 6 above)
Comment (SwRaw) [reply docs/environment_variables.rst:10]: updated.

File: docs/index.rst (hunk 5 above)
Comment (SwRaw) [reply docs/index.rst:11]: corrected.

File: docs/installation.rst (hunk 2 above)
Comment (SwRaw) [reply docs/installation.rst:18]: @dayatsin-amd I have already added "See `how to install ROCm <https://rocm.docs.amd.com/projects/install-on-linux/en/latest/>`_."
Should we require command to install ``libdrm`` package?

File: docs/installation.rst (hunk 1 above)
Comment (SwRaw) [reply docs/installation.rst:49]: @dayatsin-amd I have already added the above commands. Should we add commands to install support packages ``libelf-dev`` and ``g++`` and ``xxd``?

File: docs/sphinx/_toc.yml.in [hunk 11]
```diff
@@ -2,11 +2,20 @@
...
-    - file: environment_variables
+    - file: install/installation
+
+  - caption: API reference
+    entries:
+    - file: api-reference/c-interface-adaptors
+    - file: api-reference/environment_variables
+    - file: api-reference/api
+    
+  - caption: Contribution
```
Comment (yhuiYH) [inline docs/sphinx/_toc.yml.in:15]: Can we change "Contribution" to "How to" instead?  Same with card in index.rst?
```suggestion
  - caption: How to
```

File: docs/index.rst (hunk 3 above)
Comment (yhuiYH) [reply docs/index.rst:30]: Can we change "Contribution" to "How to" instead?  Same with card in index.rst?

File: docs/about/license.rst [hunk 12]
```diff
@@ -0,0 +1,4 @@
+License
```
Comment (yhuiYH) [inline docs/about/license.rst:1]: Actually, for license, I prefer to get rid of the "about" folder.  I'm ok with license.rst to sit in docs/license.rst.

Comment (SwRaw) [reply docs/about/license.rst:1]: done

File: docs/sphinx/_toc.yml.in (hunk 11 above)
Comment (SwRaw) [reply docs/sphinx/_toc.yml.in:15]: @yhuiYH I am not sure if contribution should be a part of how to. How to should envelope stepwise information to achieve a specific goal. Also, it is better to keep information for users and contributors separately as those are different audiences. @lpaoletti Let me know the final decision and I shall implement it.

File: docs/install/installation.rst [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+Here is how you can install ROCR using prebuilt binaries.
+
+Prerequisites
+*******************
+
+- A system supporting ROCm. See the `supported operating systems <https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html#supported-operating-systems>`_.
+
+- Install ROCm. See `how to install ROCm <https://rocm.docs.amd.com/projects/install-on-linux/en/latest/>`_.
+
+- Install ``libdrm`` package.
```
Comment (SwRaw) [inline docs/install/installation.rst:25]: @dayatsin-amd Please provide command to install libdrm if required.

File: docs/install/installation.rst [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+--------------------------------
+
+Here is how you can build ROCR from source.
+
+Prerequisites
+***************
+
+- CMake 3.7 or later. Export CMake bin into your PATH.
+
+- Support packages ``libelf-dev`` and ``g++``.
```
Comment (SwRaw) [inline docs/install/installation.rst:54]: @dayatsin-amd Provide command to install libelf-dev and g++ is required.

File: docs/install/installation.rst [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+- A compatible version of the ``libhsakmt`` library and the ``hsakmt.h`` header file. Obtain the latest version of these files from the `ROCT-Thunk-Interface repository <https://github.com/ROCm/ROCT-Thunk-Interface>`_.
+
+- Install ``xxd``.
```
[the remark this answers is not in the document]
Comment (dayatsin-amd) [reply docs/install/installation.rst:58]: sudo apt install xxd

File: docs/install/installation.rst (hunk 14 above)
Comment (dayatsin-amd) [reply docs/install/installation.rst:54]: sudo apt install libelf-dev g++

File: docs/install/installation.rst (hunk 13 above)
Comment (dayatsin-amd) [reply docs/install/installation.rst:25]: sudo apt install libdrm-dev

**Pull request merged**